app/models/miq_dialog.rb

```diff
@@ -1,30 +1,35 @@
 class MiqDialog < ApplicationRecord
-  validates_presence_of   :name, :description
-  validates_uniqueness_of :name, :scope => :dialog_type, :case_sensitive => false
-
-  DIALOG_DIR = Rails.root.join("product/dialogs/miq_dialogs")
+  validates :name, :description, :presence => true
+  validates :name, :uniqueness => { :scope => :dialog_type, :case_sensitive => false }
 
   DIALOG_TYPES = [
     [_("VM Provision"),                "MiqProvisionWorkflow"],
     [_("Configured System Provision"), "MiqProvisionConfiguredSystemWorkflow"],
     [_("Host Provision"),              "MiqHostProvisionWorkflow"],
     [_("VM Migrate"),                  "VmMigrateWorkflow"],
-  ]
+  ].freeze
 
   serialize :content
 
   def self.seed
-    sync_from_dir
+    sync_from_dir(Rails.root.join('product', 'dialogs', 'miq_dialogs'))
+    sync_from_plugins
+  end
+
+  def self.sync_from_dir(root)
+    Dir.glob(root.join("*.{yaml,yml}")).each { |f| sync_from_file(f, root) }
```

Comment:
I'm refactoring seeding of `dialog.rb` as well and noticed this loads .yaml _and_ .yml files. So should we do here too